app/code/core/Mage/Catalog/Model/Resource/Product/Indexer/Price.php

```diff
@@ -309,6 +309,10 @@ public function reindexProductIds($ids)
      */
     protected function _getIndexer($productTypeId)
     {
+        if ($productTypeId === null) {
+            $productTypeId = '';
+        }
```

Comment:
Why not like this?

`$productTypeId ??= '';`

Reply:
Thanks for suggestion, but i think there are alot similiar occurrences that could be changed.

I'd keep it and wait for a rector rule (i am sure the is one for hort opperator)